### ISSUE TITLE
direction of expansion of Pluto cells containing a <details> html elements

### ISSUE THREAD

**Issue by disberd** [fonsp/Pluto.jl#1204]
I originally posted this as a question on zulip but @fonsp asked me to submit an issue so here it is

I was looking at the [notebook](https://plutocon2021-demos.netlify.app/fonsp%20%E2%80%94%20javascript%20inside%20pluto) from the fonsp talk on using Javascript in Pluto
and I found out about the nice use of `<details>` html tag to create blocks of text that can be expanded by user click.

One issue I have with simply having the current way this works is that pluto output cells are "anchored" at the bottom, in the sense that if the cell expands the point on the window browser stays at the bottom of the output cell.
This is the opposite of what I would like to achieve for cells containing a `<details>` element as if the hidden piece of text is quite big, you are directly skipping the beginning of it and just see the bottom part of the hidden text when "showing" the element.

I uploaded a video to try and clarify better what I mean.

https://user-images.githubusercontent.com/12846528/120194880-76a0f700-c21e-11eb-9b96-fcce8e81bb2e.mp4

From the video it's clear that opening/closing the details element breaks the reading continuity.
Ideally one would want the point in the page to remain fixed on the line containing the `<summary>` html field.

I am sure there must be an easy way using html/css/js to simply change this behavior for cells that include `<details>` elements, but I have not much clue yet as to how to achieve this.

Is this something easy to fix?

**Comment by fonsp** [comment]
I believe that this is caused by Pluto's scroll compensation: when a cell changes height, Pluto will scroll the page to compensate the height difference, **if** 

> the cell output that changed height appears above the currently focused cell

The strange scrolling behaviour with `detail` might be caused by a faulty implementation, or we might need to change the compensation rule above. It could also be the case that clicking `detail` will focus the element, but clicking it a second time blurs it, causing two different behaviours.

To investigate:
1. Is this caused by scroll compensation? Disable scroll compensation in source code to figure it out
2. What should be changed to fix this behaviour?

**Comment by disberd** [comment]
@fonsp If you can point me to the part of the Pluto code where this scroll compensation is happening I can try start having a look and maybe provide some comments to help understanding this issue.
I am quite new to the whole HTML/CSS/JavaScript ordeal but I love Pluto and plan on using it extensively for most of my future work/research so I am more than happy to have some useful study cases to dig deeper into it.

**Comment by fonsp** [comment]
Search for "compensate" here: https://github.com/fonsp/Pluto.jl/blob/main/frontend/components/CellOutput.js

**Comment by disberd** [comment]
Ok I tried looking at the code you referenced and indeed the issue seems to be due to the auto-scrolling.
For what concerns the prupose of this issue of not performing auto-scrolling when opening `<details>` elements it seems, that modifying a single line is sufficient to obtain the desired behavior.

The modification is on line 31 on this function:

https://github.com/fonsp/Pluto.jl/blob/b217b782f618516caadfc45f77b35128516d52d3/frontend/components/CellOutput.js#L28-L34

where simply adding a condition to the if block to avoid the scrolling in case the current element on focus is a `<summary>` tag makes the notebook exhibit the wanted behavior without removing autoscrolling for other cases:

```js
if (!(document.activeElement.tagName == "SUMMARY") && (cell_outputs_after_focused.length == 0 || !Array.from(cell_outputs_after_focused).includes(this.base))) {
                    window.scrollBy(0, new_height - this.old_height)
                }
```

Here is a video example of the behavior with the same notebook where I just updated the code from the developer console in chrome.

https://user-images.githubusercontent.com/12846528/120938955-b9c70280-c715-11eb-98dc-ce6f0a26f5e8.mp4

This maybe goes against style guides but at least confirms the behaviour to be controlled by the scrolling
